src/app/components/mno-product-selector/mno-product-selector.controller.coffee

```diff
@@ -66,9 +71,39 @@
 
       promise.then(
         (response) ->
+          # Extract the categories
+          categories = []
+          response.data.forEach((p) ->
+            categories = categories.concat(p.categories))
+          $ctrl.categories = _.uniq(categories)
+
           $ctrl.products = orderByFilter(response.data, 'name')
+          $ctrl.filteredProducts = $ctrl.products
       ).finally(-> $ctrl.isLoadingProducts = false)
 
+    # Filter products by name or category
+    $ctrl.onSearchChange = () ->
+      $ctrl.selectedCategory = ''
+      $ctrl.filteredProducts = []
+      firstFilterResult = []
+      for product in $ctrl.products
+        if ($ctrl.searchTerm? && $ctrl.searchTerm.length > 0) || !$ctrl.selectedCategory
```

Comment:
You're doing this test on every iteration of the`for` loop but the result is not dependent on the product

Also it can be written
```coffeescript
if $ctrl.searchTerm?.length > 0 || !$ctrl.selectedCategory
```
Also the `$ctrl.selectCategory` is always `''` in this case as you reset it on L86

So the whole loop is useless unless I missed something